### ISSUE TITLE
example and screencast broken due to missing netcat in ubuntu

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#2290]
Found by a user...

```
$ docker run --name hello ubuntu netcat -lk 1234
docker: Error response from daemon: Container command 'netcat' not found or does not exist..
```

This is featured in our [running example](https://www.weave.works/docs/net/latest/using-weave/) and the screencast at https://www.weave.works/docs/net/latest/installing-weave/

Obviously this used to work, so presumably it broke with a recent change to the ubuntu image.

@errordeveloper

**Comment by bboreham** [comment]
I had also noticed that the Ubuntu image had shrunk lately - down from 187MB 4 months ago to 120MB today.

Some related discussion here: https://github.com/tianon/docker-brew-ubuntu-core/issues/48

TL;DR: Canonical have reduced the package set.  You can get one with `netcat` via `ubuntu:trusty`

**Comment by rade** [comment]
Ouch, this isn't just broken in the screencast; our running example is based on the ubuntu image containing 'nc'.

**Comment by bboreham** [comment]
You think we should use `alpine` instead of `ubuntu` ?  The `nc` syntax is slightly different.

**Comment by rade** [comment]
The alpine image doesn't have a default command, so you always have to run it with "... /bin/sh" which is incredibly annoying. Has other quirks too, like weird dns resolution. Plus nobody has heard of it. So, in short, no.

**Comment by bboreham** [comment]
We have 35 instances of `docker run ... ubuntu`, but only a couple of them seem to invoke `nc` so I'll just fix those.

**Comment by bboreham** [comment]
I changed the basic intro example to use `ubuntu:trusty` in 79d78ae97b65e225098c50aa900888106c9b8f89

Still need video re-done and, ideally, a better fix.

**Comment by bboreham** [comment]
Still needs video re-done @errordeveloper

**Comment by errordeveloper** [comment]
Would it be okay to make a new video, but keep the voiceover as is?

**Comment by bboreham** [comment]
> Would it be okay to make a new video, but keep the voiceover as is?

The voice says "using plain ubuntu image", so no, not if you want them to match.

**Comment by rade** [comment]
> The voice says "using plain ubuntu image", so no, not if you want them to match.

I think it would be acceptable. Though generally getting audio and video to sync when only recording one of them is tricky, due to commands taking different amounts of time, being able to hear the typing, etc. Depends on how perfectionist you want to be.

**Comment by errordeveloper** [comment]
> Though generally getting audio and video to sync when only recording one of them is tricky

True, but I've got this setup in the way that I can just replace strings in a JSON file and replay it with [`asciinema`](https://asciinema.org/), so if that's okay for us, it'd only take a few hours to do. I am wondering about this because there had been some discussion of re-working the whole scenario, which would obviously imply quite a bit more work, and I don't want to do something half-way or even less, unless everyone is happy with the outcome...

**Comment by bboreham** [comment]
Any update on this @errordeveloper @mjlodge ?

There is now a 'bash' image that may suit better - see #2553

**Comment by rade** [comment]
@errordeveloper hello?

**Comment by errordeveloper** [comment]
@rade I've discussed this with @lukemarsden, he'd like to take care of it.

**Comment by bboreham** [comment]
@lukemarsden how is this coming along?

**Comment by lukemarsden** [comment]
@bboreham it's not, yet. It's on the [DX backlog](https://github.com/weaveworks/dx/projects/1).

**Comment by errordeveloper** [comment]
Closing as this is now very stale.
